netty/src/test/java/io/grpc/netty/NettyServerHandlerTest.java

```diff
@@ -529,6 +533,56 @@ public void keepAliveEnforcer_noticesInactive() throws Exception {
         eq(Http2Error.ENHANCE_YOUR_CALM.code()), any(ByteBuf.class), any(ChannelPromise.class));
   }
 
+  @Test
+  public void maxConnectionAge() throws Exception {
+    maxConnectionAgeInNanos = TimeUnit.MILLISECONDS.toNanos(10L);
+    setUp();
+
+    assertTrue(channel().isOpen());
+
+    Thread.sleep(5L);
+    channel().runPendingTasks();
+
+    // GO_AWAY not sent yet
```

Comment:
Penn pointed out that scheduling anything new only happens in `runPendingTasks()`, so this would actually be reliable.